pubsub/gossipsub/gossipsub-v1.1.md

```diff
@@ -0,0 +1,405 @@
+
+In order to avoid triggering the penalty too early, the parameter has an activation window.
+This is a configurable value that is the time that the peer must have been in the mesh before
+the parameter applies.
+
+In pseudo-go:
+```go
+// topic configuration parameters
+var MeshMessageDeliveriesCap, MeshMessageDeliveriesThreshold     float64
+var MeshMessageDeliveriesWindow, MeshMessageDeliveriesActivation time.Duration
+
+// time in mesh, lazily updated
+var meshTime time.Duration
+
+// counter updated every time a first or near-first message delivery occurs by a mesh peer
+var meshMessageDeliveries float64
+
+// counter update
+meshMessageDeliveries += 1
+if meshMessageDeliveries > MeshMessageDeliveriesCap {
+  meshMessageDeliveries = MeshMessageDeliveriesCap
+}
+
+// calculation of P₃
+var deficit float64
+if meshTime > MeshMessageDeliveriesActivation && meshMessageDeliveries < MeshMessageDeliveriesThreshold {
+  deficit = MeshMessageDeliveriesThreshold - meshMessageDeliveries
+}
+
+p3 := deficit * deficit
+```
+
+In order to calculate P₃b, the router maintains a counter that is updated whenever the peer is pruned
+with an active deficit in message delivery. The parameter is uncapped.
+
+In pseudo-go:
+```go
+// counter updated at prune time
+var meshFailurePenalty float64
+
+// counter update
+if meshTime > MeshMessageDeliveriesActivation && meshMessageDeliveries < MeshMessageDeliveriesThreshold {
+  deficit = MeshMessageDeliveriesThreshold - meshMessageDeliveries
+  meshFailurePenalty += deficit * deficit
+}
+
+// P₃b
+p3b := meshFailurePenalty
+```
+
+##### P₄: Invalid Messages
+
+In order to compute P₄, the router maintains a counter that increments whenever a message fails
+validation. The counter is uncapped.
+
+In pseudo-go:
+```go
+// counter updated every time a message fails validation
+var invalidMessageDeliveries float64
+
+// counter update
+invalidMessageDeliveries += 1
+
+// P₄
+p4 := invalidMessageDeliveries
+```
+
+##### Parameter Decay
+
+The counters associated with P₂, P₃, P₃b, and P₄ decay periodically by multiplying with a configurable
+decay factor. When the value drops below a threshold it is considered zero.
+
+In pseudo-go:
+```go
+// decay factors
+var FirstMessageDeliveriesDecay, MeshMessageDeliveriesDecay, MeshFailurePenaltyDecay, InvalidMessageDeliveriesDecay float64
+
+// 0-threshold
+var DecayToZero float64
+
+// periodic decay of counters
+firstMessageDeliveries *= FirstMessageDeliveriesDecay
+if firstMessageDeliveries < DecayToZero {
+  firstMessageDeliveries = 0
+}
+
+meshMessageDeliveries *= MeshMessageDeliveriesDecay
+if meshMessageDeliveries < DecayToZero {
+  meshMessageDeliveries = 0
+}
+
+meshFailurePenalty *= MeshFailurePenaltyDecay
+if meshFailurePenalty < DecayToZero {
+  meshFailurePenalty = 0
+}
+
+invalidMessageDeliveries *= InvalidMessageDeliveriesDecay
+if invalidMessageDeliveries < DecayToZero {
+  invalidMessageDeliveries = 0
+}
+```
+
+#### Guidelines for Tuning the Scoring Function
+
+TBD
```

Comment:
@vyzo any ETA?

Reply:
once we've made some progress with the testground tests.